apps/scan/src/main/scala/org/lfdecentralizedtrust/splice/scan/store/db/DbScanVerdictStore.scala

```diff
@@ -213,24 +213,22 @@ class DbScanVerdictStore(
 
     if (items.isEmpty) Future.unit
     else {
-      val (headVerdict, _) = items.headOption.getOrElse(
-        throw new RuntimeException("insertVerdictAndTransactionViews called with empty items")
-      )
-      val checkExists = sql"""
-             select exists(
-               select 1
+      val checkExist = (sql"""
+               select update_id
                from #${Tables.verdicts}
                where history_id = $historyId
-                 and update_id = ${headVerdict.updateId}
-             )
-           """.as[Boolean].head
+                 and update_id IN """ ++ inClause(items.map(_._1.updateId))).as[String]
```

Comment:
👍 ~should~Will  use index just fine cause Oriol checked it 😄 (and this is like max 50 update_id's at a time (because of batch), so should be good)